tests/unit/managed_array_unit_tests.cpp

```diff
@@ -35,6 +35,25 @@ TEST(ManagedArray, SpaceConstructorCPU)
   array.free();
 }
 
+TEST(ManagedArray, ReallocDefaultConstructor)
+{
+  chai::ManagedArray<float> array;
+  array.reallocate(1);
+  float val = array[0];
```

Comment:
This causes a segfault in the current implementation, as the ManagedArray doesn't actually allocate anything during the reallocate call.

Reply:
changes look reasonable to me.